GTG/gtk/preferences.py

```diff
@@ -75,71 +73,64 @@ def disable_gtg_autostart():
         os.remove(AUTOSTART_PATH)
 
 
-class PreferencesDialog(object):
-    """ Show preference dialog """
+class PreferencesWindow(object):
+    """ Show preference window """
+
+    PREFERENCES_UI_FILE = os.path.join(UI_DIR, "preferences.ui")
 
     def __init__(self, req, vmanager):
         self.req = req
         self.config = self.req.get_config('browser')
         builder = Gtk.Builder()
-        builder.add_from_file(ViewConfig.PREFERENCES_UI_FILE)
-
-        self.dialog = builder.get_object("PreferencesDialog")
-        self.dialog.set_title(_("Preferences - %s" % info.NAME))
-        self.pref_autostart = builder.get_object("pref_autostart")
-        self.pref_show_preview = builder.get_object("pref_show_preview")
-        self.bg_color_enable = builder.get_object("bg_color_enable")
-        self.hbox1 = builder.get_object("hbox1")
+        builder.add_from_file(self.PREFERENCES_UI_FILE)
+
+        self.window = builder.get_object("MainWindow")
+        self.autostart_button = builder.get_object("autostart_button")
+        self.preview_button = builder.get_object("preview_button")
+        self.bg_color_button = builder.get_object("bg_color_button")
         self.shortcut_button = builder.get_object("shortcut_button")
+        self.font_button = builder.get_object("font_button")
 
         self.shortcut = ShortcutWidget(builder)
-        help.add_help_shortcut(self.dialog, "preferences")
 
-        self.fontbutton = builder.get_object("fontbutton")
         self.timer = vmanager.timer
         self.color_invalid = Color(50000, 0, 0)
-        self.refresh_hour = builder.get_object("hour")
-        self.refresh_mins = builder.get_object("min")
+        self.refresh_hour = builder.get_object("hour_entry")
+        self.refresh_mins = builder.get_object("minute_entry")
         editor_font = self.config.get("font_name")
         if editor_font == "":
-            font = self.dialog.get_style_context().get_font(
+            font = self.window.get_style_context().get_font(
                 Gtk.StateFlags.NORMAL)
             editor_font = font.to_string()
-        self.fontbutton.set_font_name(editor_font)
+        self.font_button.set_font_name(editor_font)
 
         builder.connect_signals({
```

Comment:
Hmm, however how to handle this: self.shortcut.on_shortcut_toggled?

I'm always getting attribute error on this one :/ probably because it references a different class? Is it possible to handle this case in any way?

And more importantly: what kind of magic is this? Can you provide some source that would explain why this is possible? :blush: